src/grpc_client.py

```diff
@@ -0,0 +1,148 @@
+    fun = lambda stub: stub.GetExperimentDetails(expr)
+    experiment: hpo_pb2.ExperimentDetails = run(fun)
+    json_obj = MessageToJson(experiment)
+    click.echo(json_obj)
+
+@main.command()
+@click.option("--file", prompt=" Experiment configuration file path", type=str)
+def new(file):
+    """Create a new experiment"""
+    # TODO: validate file path
+    with open(file, 'r') as json_file:
+        data = json.load(json_file)
+        # data = file.read().replace('\n', '')
+        message: hpo_pb2.ExperimentDetails = ParseDict(data, hpo_pb2.ExperimentDetails())
+    click.echo(" Adding new experiment: {}".format(message.experiment_name))
+    fun = lambda stub: stub.NewExperiment(message)
+    response: hpo_pb2.NewExperimentsReply = run(fun)
+    click.echo("Trial Number: {}".format(response.trial_number))
+
+@main.command()
+@click.option("--name", prompt=" Experiment name", type=str)
+@click.option("--trial", prompt=" Trial number", type=int)
+def config(name, trial):
+    """Obtain a configuration set for a particular experiment trail"""
+    expr: hpo_pb2.ExperimentTrial = hpo_pb2.ExperimentTrial()
+    expr.experiment_name = name
+    expr.trial = trial
+    fun = lambda stub: stub.GetTrialConfig(expr)
+    trial_config: hpo_pb2.TrialConfig = run(fun)
+    json_obj = MessageToJson(trial_config)
+    click.echo(json_obj)
+
+@main.command()
+@click.option("--name", prompt=" Enter name", type=str)
+@click.option("--trial", prompt=" Enter trial number", type=int)
+@click.option("--result", prompt=" Enter trial result", type=str)
+@click.option("--value_type", prompt=" Enter result type", type=str)
+@click.option("--value", prompt=" Enter result value", type=float)
+def result(name, trial, result, value_type, value):
+    """Update results for a particular experiment trail"""
+    trialResult: hpo_pb2.ExperimentTrialResult = hpo_pb2.ExperimentTrialResult()
+    trialResult.experiment_id = name
+    trialResult.trial = trial
+    trialResult.result = hpo_pb2._EXPERIMENTTRIALRESULT_RESULT.values_by_name[result].number
+    trialResult.value_type = value_type
+    trialResult.value = value
+    fun = lambda stub: stub.UpdateTrialResult(trialResult)
+    hpo_pb2.TrialConfig = run(fun)
+    click.echo("Success: Updated Trial Result")
+
+@main.command()
+@click.option("--name", prompt=" Enter name", type=str)
+def next(name):
+    """Generate next configuration set for running experiment"""
+    experiment: hpo_pb2.ExperimentNameParams = hpo_pb2.ExperimentNameParams()
+    experiment.experiment_name = name
+    fun = lambda stub : stub.GenerateNextConfig(experiment)
+    reply: hpo_pb2.NewExperimentsReply = run(fun)
+    click.echo("Next Trial: {}".format(reply.trial_number))
+
+def run(func):
+    # NOTE(gRPC Python Team): .close() is possible on a channel and should be
+    # used in circumstances in which the with statement does not fit the needs
+    # of the code.
+    with grpc.insecure_channel('localhost:50051') as channel:
```

Comment:
Can we have the IP and PORT defined at the top and used everywhere

Reply:
Fixed in [896cb0e](https://github.com/kruize/hpo/pull/21/commits/896cb0e12348ec9e34612760df80034e7b3b33b9) . Ideally this (and the REST port/interface) should be configurable